test/test_python_dispatch.py

```diff
@@ -523,5 +523,51 @@ def wrap(e):
         self.assertEqual(x.grad, y)
         self.assertEqual(y.grad, x)
 
+    def test_none_wrapping(self):
+        # A Tensor subclass that returns None when doing add
+        # See LoggingTensor above for more details on the subclass
+        class SublassWithNone(torch.Tensor):
+            @staticmethod
+            def __new__(cls, elem, *args, **kwargs):
+                r = torch.Tensor._make_wrapper_subclass(
+                    cls, elem.size(),
+                    dtype=elem.dtype, layout=elem.layout,
+                    device=elem.device, requires_grad=elem.requires_grad
+                )
+                r.elem = elem
+                return r
+
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+                def unwrap(e):
+                    return e.elem if isinstance(e, SublassWithNone) else e
+
+                def wrap(e):
+                    return SublassWithNone(e) if isinstance(e, torch.Tensor) else e
+
+                # no_dispatch is only needed if you use enable_python_mode.
+                # It prevents infinite recursion.
+                with no_dispatch():
+                    rs = tree_map(wrap, func(*tree_map(unwrap, args), **tree_map(unwrap, kwargs)))
+                if func.__name__ == "add":
+                    return None
+                else:
+                    return rs
+
+        x = SublassWithNone(torch.rand(2))
```

Comment:
After this change, what happens with the following?
```
torch.add(torch.randn(3), None)
```
If I am understanding this PR correctly, the above should still raise an error. Is that a correct assessment? (If that is correct then I don't have any concerns with this PR and will leave the JIT parts to Elias and co)

Reply:
It does not raise in arg parsing but it does raise in the VariableType kernel when we call `unpack`. Just like today.